### ISSUE TITLE
[ui_state] Not working properly

### ISSUE THREAD

**Issue by Vzz1c** [fzyzcjy/flutter_rust_bridge#2440]
```dart
Future<void> main() async {
 return runRustApp(body: body, state: RustState.newInstance);
}

Widget body(Future<RustState> futureState) {
  return FutureBuilder<RustState>(
    future: futureState,
    builder: (context, snapshot) {
      if (snapshot.connectionState == ConnectionState.waiting) {
        return CircularProgressIndicator();
      } else if (snapshot.hasError) {
        return Text('Error: ${snapshot.error}');
      } else if (snapshot.hasData) {
        RustState state = snapshot.data!;
        return Column(
          children: [
            Text('Count: ${state.count}'),
            TextButton(onPressed: state.increment, child: Text('+1')),
          ],
        ).padding(all: 16);
      } else {
        return Text('No data available');
      }
    },
  );
}
```
`Class 'Future<RustState>' has no instance method 'setBaseState'.
Receiver: Instance of 'Future<RustState>'
Tried calling: setBaseState(baseState: Instance of 'BaseRustStateImpl')`

**Comment by fzyzcjy** [comment]
Hi, according to the demo https://github.com/fzyzcjy/flutter_rust_bridge/blob/ebb6a27836b4669276a9f76e6e31a1047a28f093/frb_example/rust_ui_counter/ui/lib/main.dart#L8, it seems we should use

```
Widget body(RustState state) {
```

instead of 

```
Widget body(Future<RustState> futureState) {
```

**Comment by Vzz1c** [comment]
The code I generated using the latest version of flutter_rust_bridge_codegen
```dart
abstract class RustState implements RustOpaqueInterface {
  int get count;

  set count(int count);

  Future<void> increment();

  // HINT: Make it `#[frb(sync)]` to let it become the default constructor of Dart class.
  static Future<RustState> newInstance() =>
      RustLib.instance.api.crateApiSimpleRustStateNew();

  Future<void> setBaseState({required BaseRustState baseState});
}
```
`frb_example/rust_ui_counter/ui/lib/src/rust/app.dart`
```dart
abstract class RustState implements RustOpaqueInterface {
  int get count;

  set count(int count);

  void increment();

  factory RustState() => RustLib.instance.api.crateAppRustStateNew();

  void setBaseState({required BaseRustState baseState});
}
```

**Comment by fzyzcjy** [comment]
Not very sure about your question, but I mean your own code maybe need to remove that `Future` like mentioned above.

**Comment by Vzz1c** [comment]
Please delete my question completely, thank you.

**Comment by fzyzcjy** [comment]
No worries, everyone asks naive questions when starting programming!

**Comment by Vzz1c** [comment]
` naive questions`?Have you tested the code yourself? Run the `great` code: cargo install `flutter_rust_bridge_codegen && flutter_rust_bridge_codegen create my_app && cd my_app && flutter run`, run `flutter_ rust_bridge/frb_example/rust_ui_counter/src
/app.rs` example, do you get it to work? The code doesn't even match.

**Comment by fzyzcjy** [comment]
> Have you tested the code yourself?

Yes, indeed both manually and automatically (via CI). The CI tests the code for each commit, e.g. https://github.com/fzyzcjy/flutter_rust_bridge/blob/3eeaa9694f84df296a93a837a60d7cafaf6746af/.github/workflows/ci.yaml#L435.

If you find any issues running the command, it is suggested to create an issue with detailed error logs or minimal reproducible sample.

> Run the `great` code

I never claim my code is "great". It is because you mentioned deleting your question completely that I guess you thought your question was too naive and I hope to make you not felt that, thus I said the "naive question" thing. In addition, it is true that new learners often ask naive questions, and some people even claim [there are no dumb questions](https://academia.stackexchange.com/questions/51631/how-to-ask-dumb-questions).

> do you get it to work? The code doesn't even match.

It is confusing what do you mean by "match". Surely the quickstart and the rust_ui_counter are different demos, so the code should be different.

**Comment by Vzz1c** [comment]
If only you had written the code with half the focus you have in your replies, you don't even bother to validate the questions, there is no point in picking on my words, I translated it through a machine, are you fighting with the translator? Good luck to you and your project!

**Comment by fzyzcjy** [comment]
> If only you had written the code with half the focus you have in your replies

To be honest, I spent a large amount of time writing the code - which can be clearly seen from git commit history, much more than handling GitHub replies.

> you don't even bother to validate the questions

It seems I have already answer your question [here](https://github.com/fzyzcjy/flutter_rust_bridge/issues/2440#issuecomment-2525875833), and I am not sure what your further [reply](https://github.com/fzyzcjy/flutter_rust_bridge/issues/2440#issuecomment-2526551065) asks (thus I said "not very sure about your question in the latest reply ..."). Maybe add a few sentences describing what makes you confused may help. For example, what error do you see, etc. Only pasting two code snippets and expect a busy and tired programmer (me) to guess what you mean may not seem to super ideal.

> there is no point in picking on my words

I am not picking on any words. As mentioned above, since you mentioned you wanted to delete the whole question, I am worried whether you felt the question is too easy and thus want to remove it from your thread history, and thus wanted to comfort you. Not sure what your machine translation convert my sentences to.

**Comment by Vzz1c** [comment]
I hope you stay confident.

**Comment by fzyzcjy** [comment]
Close since inactivity, but feel free to reopen if you have any questions or reproducible samples!
